stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/listener/internal/SendMessageListenerState.kt

```diff
@@ -52,6 +53,8 @@ internal class SendMessageListenerState(private val logic: LogicRegistry) : Send
         channelId: String,
         message: Message,
     ) {
+        handleLastSentMessageDate(result, message)
+
         val cid = "$channelType:$channelId"
 
         if (logic.getMessageById(message.id)?.syncStatus == SyncStatus.COMPLETED) return
```

Comment:
Sometimes `logic.getMessageById(message.id)?.syncStatus` returns `SyncStatus.COMPLETED` here as for some reason `MessageNewEvent` arrives before this method is executed and overwrites `IN_PROGRESS` or `AWAITING_ATTACHMENTS` statuses with `COMPLETED`.